src/class-soft-transient.php

```diff
@@ -0,0 +1,177 @@
+		$this->cron_hook = $hook;
+		return $this;
+	}
+
+	/**
+	 * Set the arguments to pass to the cron event.
+	 *
+	 * @param array<mixed> $args The arguments to pass to the cron event.
+	 * @return self
+	 */
+	public function set_cron_args( array $args ): self {
+		$this->cron_args = $args;
+		return $this;
+	}
+
+	/**
+	 * Get the arguments to pass to the cron event.
+	 *
+	 * @return array<mixed>
+	 */
+	public function get_cron_args(): array {
+		return array_merge( [ $this->key ], $this->cron_args );
+	}
+
+	/**
+	 * Get a "soft" transient. This is a transient which is updated via wp-cron,
+	 * so refreshing the transient doesn't slow the request. Otherwise, this works
+	 * just like get_transient().
+	 *
+	 * If the transient does not exist or does not have a value, then the return value
+	 * will be false.
+	 *
+	 * @return mixed Value of transient.
+	 */
+	public function get() {
+		$transient = \get_transient( $this->key );
+		if ( false === $transient ) {
+			return false;
+		}
+
+		// Ensure that this is a soft transient.
+		if ( ! is_array( $transient ) || empty( $transient['expiration'] ) || ! array_key_exists( 'data', $transient ) ) {
+			return $transient;
+		}
+
+		// Check if the transient is expired.
+		$expiration = intval( $transient['expiration'] );
+		if ( ! empty( $expiration ) && $expiration <= time() ) {
+			// Cache needs to be updated.
+			if ( ! empty( $transient['status'] ) && (
+				'ok' === $transient['status']
+				|| (
+					// If the transient is still loading, it's been an hour since it expired, and
+					// the cron event isn't scheduled, then schedule it again.
+					'loading' === $transient['status']
+					&& $expiration <= time() - HOUR_IN_SECONDS
+					&& ! \wp_next_scheduled( $this->get_cron_hook(), $this->get_cron_args() )
+				)
+			) ) {
+				// Schedule the update event.
+				\wp_schedule_single_event( time(), $this->get_cron_hook(), $this->get_cron_args() );
+				$transient['status'] = 'loading';
+
+				// Update the transient to indicate that we've scheduled a reload.
+				\set_transient( $this->key, $transient );
+
+				/**
+				 * Fires when a soft transient is scheduled to refresh.
+				 *
+				 * @param string         $transient_key The key of the transient.
+				 * @param Soft_Transient $transient     This Soft_Transient object.
+				 */
+				\do_action( 'soft_transients_expiration', $this->key, $this );
+			}
+		}
+
+		return $transient['data'];
+	}
+
+	/**
+	 * Set/update the value of a "soft" transient. This is a transient that, when
+	 * it expires, will continue to return the value and refresh via wp-cron.
+	 *
+	 * You do not need to serialize values. If the value needs to be serialized, then
+	 * it will be serialized before it is set.
+	 *
+	 * @param mixed $value      Transient value. Must be serializable if non-scalar. Expected to
+	 *                          not be SQL-escaped.
+	 * @param int   $expiration Optional. Time until expiration in seconds, default 0.
```

Comment:
maybe a note that if this is not provided, it will be a regular transient and not a soft transient?